src/Csv/ColumnsParser.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\AzureCostExtractor\Csv;
+
+use Keboola\AzureCostExtractor\Config;
+use Keboola\AzureCostExtractor\Exception\UnexpectedColumnException;
+use Keboola\Datatype\Definition\BaseType;
+
+class ColumnsParser
+{
+    private Config $config;
+
+    private array $groupingDimensions;
+
+    public function __construct(Config $config)
+    {
+        $this->config = $config;
+        $this->groupingDimensions = $config->getGroupingDimensions();
+    }
+
+    public function parse(array $response): array
+    {
+        // Map columns to objects
+        $columns = [];
+        foreach ($response['properties']['columns'] as $index => $data) {
+            $name = $data['name'];
+            $category = $this->parseCategory($name);
+            $rawType = $data['type'];
+            $type = $this->parseType($rawType);
+            $columns[] = new Column($index, $name, $type, $category);
+        }
+
+        // Sort columns by category
+        $this->sortColumns($columns);
+
+        return $columns;
+    }
+
+    private function parseType(string $rawType): string
+    {
+        switch ($rawType) {
+            case 'Number':
+                return BaseType::NUMERIC;
+            default:
+                return BaseType::STRING;
+        }
+    }
+
+    private function parseCategory(string $name): string
+    {
+        if ($name === 'Currency') {
+            return Column::CATEGORY_CURRENCY;
+        }
+
+        if ($name === $this->config->getAggregation()) {
+            return Column::CATEGORY_AGGREGATION;
+        }
+
+        if (in_array($name, $this->groupingDimensions, true)) {
+            return Column::CATEGORY_GROUPING_DIMENSION;
+        }
+
+        if (strpos($name, 'Date') !== false ||
+            strpos($name, 'Month') !== false
+        ) {
+            return Column::CATEGORY_TIME_DIMENSION;
+        }
+
+        throw new UnexpectedColumnException(sprintf('Found unexpected column "%s" in the API response.', $name));
+    }
+
+    private function sortColumns(array &$columns): void
```

Comment:
Poradie stlpcov z API je tiez divne, ... tu sa to zoradzuje, najprv. datum, potom group by stlpce a potom cena a currency.